pkg/volume/glusterfs/glusterfs.go

```diff
@@ -255,8 +260,17 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 	for hostIP := range addr {
 		errs = b.mounter.Mount(hostIP+":"+b.path, dir, "glusterfs", options)
 		if errs == nil {
+			glog.Infof("glusterfs: successfully mounted %s", dir)
 			return nil
 		}
 	}
+
+	// Failed mount scenario.
+	// Since gluster does not return eror text
+	// it all goes in a log file, we will read the log file
+	logerror := readGlusterLog(log, b.pod.Name)
+	if logerror != nil {
```

Comment:
It's a convention in go to re-use an error in a case like this where a list of errors is being accumulated.

Also, I think the gluster log has to be a best-effort attempt here.  There are legitimate reasons that this fail might not be present when this fails.

@thockin @saad-ali any thoughts on this?